tests/ui/components/results/paginationcomponent.js

```diff
@@ -145,6 +145,6 @@ describe('rendering the page numbers', () => {
     const next = wrapper.find('.js-yxt-Pagination-next');
     expect(next).toHaveLength(1);
     next.first().simulate('click');
-    expect(paginate).toHaveBeenCalled();
```

Comment:
Do we have test coverage for all the different edge cases that can occur in _allocate? That's one of the more complex methods in the Pagination component. We'll want to make sure we have coverage for the case in which _allocate will distribute pages evenly about the middle page and when it will be uneven since it's hit a boundary in some direction.

Reply:
I've added some unit tests, it's a bit of a PITA to test comprehensively using enzyme, but I'll see about more thoroughly checking this.